test/conformance/protocol_test.go

```diff
@@ -0,0 +1,196 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/pkg/test/spoof"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	serviceresourcenames "github.com/knative/serving/pkg/reconciler/v1alpha1/service/resources/names"
+	"github.com/knative/serving/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type protocolsTest struct {
+	t       *testing.T
+	logger  *logging.BaseLogger
+	clients *test.Clients
+	names   test.ResourceNames
+}
+
+func (pt *protocolsTest) setup(t *testing.T) {
+	pt.t = t
+
+	pt.clients = setup(t)
+	pt.logger = logging.GetContextLogger(t.Name())
+	pt.names = test.ResourceNames{
+		Service: test.AppendRandomString("protocols", pt.logger),
+		Image:   protocols,
+	}
+
+	test.CleanupOnInterrupt(func() { pt.teardown() }, pt.logger)
+}
+
+func (pt *protocolsTest) teardown() {
+	tearDown(pt.clients, pt.names)
+}
+
+func (pt *protocolsTest) getProtocol(resp *spoof.Response) protocol {
+	var got protocol
+
+	err := json.Unmarshal(resp.Body, &got)
+	if err != nil {
+		pt.t.Fatalf("Can't unmarshal response %s: %v", resp.Body, err)
+	}
+
+	pt.logger.Infof("Parsed version: %q", got.String())
+
+	return got
+}
+
+func (pt *protocolsTest) makeRequest(domain string) *spoof.Response {
+	pt.logger.Infof("Making request to %q", domain)
```

Comment:
Do we need to actually make an h2c request here to verify it works end-to-end?

Reply:
Will be covered with #3010